src/NodeApi/Runtime/NodeJSRuntime.cs

```diff
@@ -0,0 +1,2277 @@
+        napi_value msg,
+        out napi_value result)
+    {
+        if (napi_create_type_error == null)
+        {
+            napi_create_type_error = (delegate* unmanaged[Cdecl]<
+                napi_env, napi_value, napi_value, nint, napi_status>)
+                GetExport(nameof(napi_create_type_error));
+        }
+
+        result = default;
+        fixed (napi_value* result_ptr = &result)
+        {
+            return napi_create_type_error(env, code, msg, (nint)result_ptr);
+        }
+    }
+
+    private delegate* unmanaged[Cdecl]<napi_env, napi_value, napi_value, nint, napi_status>
+        napi_create_range_error;
+
+    public override napi_status CreateRangeError(
+        napi_env env,
+        napi_value code,
+        napi_value msg,
+        out napi_value result)
+    {
+        if (napi_create_range_error == null)
+        {
+            napi_create_range_error = (delegate* unmanaged[Cdecl]<
+                napi_env, napi_value, napi_value, nint, napi_status>)
+                GetExport(nameof(napi_create_range_error));
+        }
+
+        result = default;
+        fixed (napi_value* result_ptr = &result)
+        {
+            return napi_create_range_error(env, code, msg, (nint)result_ptr);
+        }
+    }
+
+    private delegate* unmanaged[Cdecl]<napi_env, napi_value, napi_value, nint, napi_status>
+        node_api_create_syntax_error;
+
+    public override napi_status CreateSyntaxError(
+        napi_env env,
+        napi_value code,
+        napi_value msg,
+        out napi_value result)
+    {
+        if (node_api_create_syntax_error == null)
+        {
+            node_api_create_syntax_error = (delegate* unmanaged[Cdecl]<
+                napi_env, napi_value, napi_value, nint, napi_status>)
+                GetExport(nameof(node_api_create_syntax_error));
+        }
+
+        result = default;
+        fixed (napi_value* result_ptr = &result)
+        {
+            return node_api_create_syntax_error(env, code, msg, (nint)result_ptr);
+        }
+    }
+
+    private delegate* unmanaged[Cdecl]<napi_env, napi_value, napi_status>
+        napi_throw;
+
+    public override napi_status Throw(napi_env env, napi_value error)
+    {
+        if (napi_throw == null)
+        {
+            napi_throw = (delegate* unmanaged[Cdecl]<napi_env, napi_value, napi_status>)
+                GetExport(nameof(napi_throw));
+        }
+
+        return napi_throw(env, error);
+    }
+
+    private delegate* unmanaged[Cdecl]<napi_env, nint, nint, napi_status> napi_throw_error;
+
+    public override napi_status ThrowError(napi_env env, string? code, string msg)
+    {
+        if (napi_throw_error == null)
+        {
+            napi_throw_error = (delegate* unmanaged[Cdecl]<napi_env, nint, nint, napi_status>)
+                GetExport(nameof(napi_throw_error));
+        }
+
+#if NET5_0_OR_GREATER
+        Span<byte> codeBytes = stackalloc byte[Encoding.UTF8.GetByteCount(code ?? string.Empty)];
```

Comment:
We must be careful with such unbounded stack allocations like this. They can lead to stack corruption and related security risks. It would be much better to do the allocations up to certain limit, and then use heap for larger strings. The Lib import generated code uses a special class to do it.

Reply:
Yes, good point. I'll fix it. (This code was refactored from `JSValue.cs`, so the issue existed there already.)

Reply:
Fixed, using `ArrayPool` and a new `PooledBuffer` class that makes it convenient to lease the pooled buffers via `using`.

Now even short string are using pooled buffers rather than `stackalloc`. I couldn't work out a way to use either `stackalloc` or pooled (heap) buffers depending on length, without making the code very messy, due to the restriction on passing fixed or `Span<>` values into a lambda.